cmd/statusd/utils.go

```diff
@@ -1397,3 +1394,39 @@ func testValidateNodeConfig(t *testing.T, config string, fn func(common.APIDetai
 
 	fn(resp)
 }
+
+// ensureNodeSync waits until synchronzation is done and more safe
+// than only waiting a fixed amount of time.
+func ensureNodeSync(t *testing.T, nodeManager common.NodeManager) {
```

Comment:
I would change a signature to `func ensureNodeSync(nodeManager common.NodeManager) error` and then did:
```go
s.NoError(ensureNodeSync(nodeManager))
```

Reply:
Also, can we move this somewhere so that we don't duplicate code here and in `e2e/suites.go`?

Reply:
Yes, removing that would be the next step. Nice idea with returning an error, would help suites as well as the standalone tests.

Reply:
What is this? You had better unify these two implementations:
https://github.com/status-im/status-go/blob/2159711fa38442254f5af8f31d943c1271e20762/e2e/suites.go#L26
https://github.com/status-im/status-go/blob/2159711fa38442254f5af8f31d943c1271e20762/e2e/suites.go#L168
instead of adding another one.

The one already merged is also supposedly more tested than yours. If you think not, let's discuss before changing, it's definitely out of scope of this issue.

Reply:
I'm currently harmonizing all to only one located in package `testing`. The algorithm is based on the hint by IMHO @adambabik and works pretty good while I had still `no suitable peers` with the one merged with the latest `develop`. After change no more peer errors.

The location of the algorithm in only one function will at least allow a quick exchange with the current one for evaluation purposes.

Reply:
Ok, got it, let's see.